docs/resource/README.md

```diff
@@ -105,7 +105,30 @@ as specified in the [Resource SDK specification](https://github.com/open-telemet
 
-**[2]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 RFC 4122 UUID (services aiming for reproducible UUIDs may also use Version 5, see RFC 4122 for more recommendations).
+**[2]:** MUST be unique for each instance of the same `service.namespace,service.name` pair (in other words `service.namespace,service.name,service.instance.id` triplet MUST be globally unique). The ID helps to distinguish instances of the same service that exist at the same time (e.g. instances of a horizontally scaled service). It is preferable for the ID to be persistent and stay the same for the lifetime of the service instance, however it is acceptable that the ID is ephemeral and changes during important lifetime events for the service (e.g. service restarts). If the service has no inherent unique ID that can be used as the value of this attribute it is recommended to generate a random Version 1 or Version 4 [RFC 4122](https://www.ietf.org/rfc/rfc4122.txt) UUID. Services aiming for reproducible UUIDs may also use Version 5.
+SDKs are required to follow the following algorithm when generating `service.instance.id`:
+- If the user has provided a `service.instance.id`, via environment
+  variable, configuration or custom resource detection, this will
+  always be used and honored over generated IDs.
+- When any of the below combinations of resource attribute are provided, they should be used as the input
+  for generating a UUID v5. The values within each combination should be separated with dots:
+  * `container.id`
+  * `k8s.namespace.name`/`k8s.pod.name`/`k8s.container.name`
+  * `host.id`
+- When the SDK can detect it's running on Kubernetes or derivatives, such as by checking the
+  availability of a directory named `/var/run/secrets/kubernetes.io`, it should use
+  the contents of the file `/var/run/secrets/kubernetes.io/serviceaccount/namespace`,
+  along with the hostname and process ID (PID) separated with a dot ("namespace.hostname.pid") as the input
+  for generating a UUID v5.
+- When the SDK is running in an environment where a `/etc/machine-id` (see MACHINE-ID(5))
```

Comment:
Is MACHINE-ID(5) intended to be a link to man page?

Reply:
Yes! I added the link to the man page.